src/linux/disk.rs

```diff
@@ -108,8 +130,9 @@ impl DiskExt for Disk {
     fn update(&mut self) -> bool {
         unsafe {
             let mut stat: statvfs = mem::zeroed();
-            if statvfs(self.mount_point.as_ptr() as *const i8, &mut stat) == 0 {
-                self.available_space = stat.f_bsize as u64 * stat.f_bavail as u64;
+            let mount_point_cpath = to_cpath(&self.mount_point);
```

Comment:
A single heap allocation is generally cheaper than the syscalls we're making in the immediate surroundings here, no?